src/Commands/core/ArchiveDumpCommands.php

```diff
@@ -0,0 +1,591 @@
+     *
+     * @throws \Exception
+     */
+    private function isWebRootSite(): bool
+    {
+        return self::WEB_DOCROOT === basename($this->siteAliasManager()->getSelf()->root());
+    }
+
+    /**
+     * Returns site's docroot name.
+     *
+     * @return string
+     *
+     * @throws \Exception
+     */
+    private function getDocrootName(): string
+    {
+        return $this->isWebRootSite() ? self::WEB_DOCROOT : '';
+    }
+
+    /**
+     * Creates "code" archive component and returns the absolute path.
+     *
+     * @param array $options
+     *  The command options.
+     *
+     * @return string
+     *  The full path to the code archive component directory.
+     *
+     * @throws \Exception
+     */
+    private function getCodeComponentPath(array $options): string
+    {
+        $codePath = $this->isWebRootSite()
+            ? dirname($this->siteAliasManager()->getSelf()->root())
+            : $this->siteAliasManager()->getSelf()->root();
+        $codeArchiveComponentPath = Path::join($this->archiveDir, self::COMPONENT_CODE);
+
+        $this->logger()->info(
+            dt(
+                'Copying code files from !from_path to !to_path...',
+                ['!from_path' => $codePath, '!to_path' => $codeArchiveComponentPath]
+            )
+        );
+
+        $excludes = $options['exclude-code-paths']
+            ? $this->getRegexpsForPaths(explode(',', $options['exclude-code-paths']))
+            : [];
+        $excludes = array_merge(
+            $excludes,
+            $this->getRegexpsForPaths(
+                [
+                    '.git',
+                    'vendor',
+                ]
+            ),
+            $this->getDrupalExcludes()
+        );
+
+        if ($this->isWebRootSite()) {
+            $excludes = array_merge(
+                $excludes,
+                $this->getWebDocrootDrupalExcludes()
+            );
+        }
+
+        $this->filesystem->mirror(
+            $codePath,
+            $codeArchiveComponentPath,
+            $this->getFileIterator($codePath, $excludes)
+        );
+
+        return $codeArchiveComponentPath;
+    }
+
+    /**
+     * Creates "Drupal files" archive component and returns the absolute path.
+     *
+     * @return string
+     *  The full path to the Drupal files archive component directory.
+     *
+     * @throws \Exception
+     */
+    private function getDrupalFilesComponentPath(): string
+    {
+        $process = $this->processManager()->drush(
+            $this->siteAliasManager()->getSelf(),
+            'core-status',
+            [],
+            ['fields' => 'files', 'format' => 'json']
+        );
+        $process->mustRun();
+        $status = $process->getOutputAsJson();
+        if (!isset($status['files'])) {
+            throw new Exception('Failed to get path to Drupal files directory');
+        }
+
+        $drupalFilesPath = Path::join($this->siteAliasManager()->getSelf()->root(), $status['files']);
```

Comment:
Use Drupal API directly to get path to files. `\Drupal::service('file_system')->realpath('public://');`

Reply:
I took this (`core-status`-based) approach from other core commands

Reply:
`core-status` needs the site alias manager for other things; archive dump does not. Using bootstrap manager instead of alias manager decreases the scope of the command, which is good.

Reply:
Sure, will use Drupal API directly. I was referring to things like https://github.com/drush-ops/drush/blob/11.x/src/Commands/sql/SqlSyncCommands.php#L117

Reply:
@greg-1-anderson Update to use bootstrapped Drupal core